crates/vm/levm/docs/faq.md

```diff
@@ -0,0 +1,45 @@
+# FAQ
+## `usize` and `U256`
+In Rust, **accessing an index on a specific data structure requires a `usize` type variable.** This can be seen in methods like `get` and `get_mut`.
+
+<!-- TODO: Link in the documentation where the `U256` adresses are described -->
+On the other hand, the EVM specification requires all addresses to be in `U256`. Therefore, every opcode treats its arguments as `U256` values.
+The problem arises in the opcodes that need to acess a specific index on a data structure (e.g. `CALLDATA`, `CODECOPY`, `EXTCODECOPY`, etc).
+These operands receive offsets and indexes in `U256`, but the data structure they have to access (e.g. `Memory` or  `Calldata`) **require a `usize`**. Therefore, those paramenters need to be cast from `U256` to `usize`.
+The problem is, `U256`'s representation ranger is larger than `usize`'s; so not all numbers can be successfuly cast. In these cases, special attention is needed.
+
+The main way to deal with theses cases (at least, at the time of writing) is to **cast the value only when you know it can fit**. Before casting to `usize`, we compare the size of the index in `U256` with the length of the datastructure it wants to access. Here's an example from the `EXTCODECOPY` opcode (NOTE: the code snippet is a simplified/altered version to demonstrate this pattern. The actual implementation is fairly different):
```

Comment:
```suggestion
The main way to deal with these cases (at least, at the time of writing) is to **cast the value only when you know it can fit**. Before casting to `usize`, we compare the size of the index in `U256` with the length of the datastructure it wants to access. Here's an example from the `EXTCODECOPY` opcode (NOTE: the code snippet is a simplified/altered version to demonstrate this pattern. The actual implementation is fairly different):
```